src/helpers.rs

```diff
@@ -480,31 +480,46 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         self.eval_context_mut().memory.write_bytes(scalar, bytes.iter().copied().chain(iter::once(0u8)))?;
         Ok(true)
     }
+
+    fn alloc_os_str_as_c_str(
+        &mut self,
+        os_str : &OsStr
+    ) -> Pointer<Tag> {
+        let bytes = os_str_to_bytes(os_str).unwrap();
+        let size = bytes.len() as u64 + 1; // Make space for `0` terminator.
+
+        let this = self.eval_context_mut();
+
+        let arg_type = this.tcx.mk_array(this.tcx.types.u8, size);
+        let arg_place = this.allocate(this.layout_of(arg_type).unwrap(), MiriMemoryKind::Env.into());
```

Comment:
The memory kind should probably be passed in by the caller; I am not sure if `Env` will always be right.